src/storage/store/engine.rs

```diff
@@ -785,7 +784,44 @@ impl ShardEngine {
                             err.to_string()
                         );
                     }
-                    last_block_event_seqnum += 1;
+
+                    if version.is_enabled(ProtocolFeature::StorageLending) {
+                        // Process storage lend messages from block events
+                        if let Some(block_event_data) = &block_event.data {
+                            if let Some(proto::block_event_data::Body::LendStorageEventBody(
+                                lend_storage_event,
+                            )) = &block_event_data.body
+                            {
+                                if let Some(lend_storage_message) =
+                                    &lend_storage_event.lend_storage_message
+                                {
+                                    match self
+                                        .stores
+                                        .storage_lend_store
+                                        .merge(lend_storage_message, txn_batch)
+                                    {
+                                        Ok(hub_event) => {
+                                            merged_messages_count += 1;
+                                            self.update_trie(trie_ctx, &hub_event, txn_batch)?;
```

Comment:
We'd be inserting this into the trie for the lender fid right? That seems wrong? I think it's more correct to insert into the borrower fid as a record of have received the lent storage.